packages/runtime-core/src/scheduler.ts

```diff
@@ -50,10 +50,12 @@ let currentFlushPromise: Promise<void> | null = null
 const RECURSION_LIMIT = 100
 type CountMap = Map<SchedulerJob, number>
 
-export function nextTick<T = void>(
+export function nextTick(): Promise<void>
+export function nextTick<T = void, R = any>(
   this: T,
-  fn?: (this: T) => void
-): Promise<void> {
+  fn: (this: T) => R
+): Promise<R extends Promise<infer P> ? P : R>
+export function nextTick<T, R>(this: T, fn?: (this: T) => R) {
```

Comment:
We can simplify this as:
```ts
export function nextTick<T = void, R = void>(
  this: T,
  fn?: (this: T) => R
): Promise<Awaited<R>> {
```

Reply:
However, `Awaited` is introduced in TypeScript 4.5... Maybe we can copy it

```ts
type Awaited<T> = T extends null | undefined
  ? T // special case for `null | undefined` when not in `--strictNullChecks` mode
  : T extends object & { then(onfulfilled: infer F, ...args: infer _): any } // `await` only unwraps object types with a callable `then`. Non-object types are not unwrapped
  ? F extends (value: infer V, ...args: infer _) => any // if the argument to `then` is callable, extracts the first argument
    ? Awaited<V> // recursively unwrap the value
    : never // the argument to `then` was not callable
  : T // non-object or non-thenable

```

Reply:
Thank you for your guidance, I have made the modifications and we can use 'Awaited' directly.

Reply:
Hmmm but `Awaited` is not supported in TypeScript < 4.5, so it will break in userland.

Reply:
Hello, I copied the `Awaited<T>` type in `shared/src/typeUtils.ts` to prevent users with TypeScript < 4.5 from encountering an unsupported `Awaited<T>` type.